Adding a previous release section

aepfli opened the pull request:
The previous release section contains all releases, which are hidden, and are not marked as draft (see 1.0.x).

relates to: #1494

Signed-off-by: Simon Schrottner <simon.schrottner@dynatrace.com>

Sidenote: this is not 100% what is describe within #1494 but covers the basic overview page for older releases. As soon as somebody sets a page to `hide: true` it will be part of the previous releases.

aepfli commented:
previous release overview 

![image](https://user-images.githubusercontent.com/9987394/205281880-22a0bfc6-3c9c-4dfc-b76c-1b127580383a.png)

overview when a previous release was selected: - also it reappears in the menu, while selected, so it is easier to navigate

![image](https://user-images.githubusercontent.com/9987394/205282047-78d899eb-8755-46e3-9298-c2473fe36b2e.png)

thisthat reviewed (approved):
Thank you @aepfli 🙌

aepfli commented:
at this stage, i removed once the root folders, so we can ensure that the build is working with those missing files - aka the previous-releases are linked properly.

we have 3 errors, but they are related to other directories on the page, also those errors are legit see https://github.com/keptn/keptn.github.io/actions/runs/3628635631/jobs/6119881919

in the next commit I will re add those files, and we should have a green build, and ready for the migration for 1.0.x on Monday (aka moving it to the root folder)

StackScribe commented:
I'm having serious reservations about this approach.  By repurposing the hide: parameter, we are now displaying links for all the ancient releases which I don't think we want to do.  And instead of a cleaner landing page, we have a landing page that is orders of magnitude messier than it was.  What does everyone else think?  I can achieve the structure I want by moving the source files around, which is not the ideal solution.  Does anyone have a better idea?

aepfli commented:
i am not sure what you mean with the landing page is messier, we only added a link in the sidebar for previous releases, nothing more. and that is anyway the desired outcome.

When the user visits the documentation, through the page, he sees only what he needs to see, nothing more nothing less. Like the state is. All the information for 1.x will be in a later stage below the root and it will be even clearer for the user to be seen.

Yes we might add KLT later - but if this will be part of Keptn docs directly, as a separate 1st level page, or as an external page is something we need to evaluate first.

Moving the source files breaks all the social media efforts and back links generate on different pages etc. and is from a SEO and traffic perspective the worst we can do. Yes we could create aliases for all the pages, but this is in a magnitude way more effort.

I agree that a documentation should be structured and easy to maintain. But I also see a website as a living document, which should preserve usability and user experience and for that fact can have flaws in favor of user experience.

aepfli commented:
for more clarity added a different flag then hide for define a previous release

StackScribe commented on content/docs/1.0.x/_index.md:
```diff
@@ -5,6 +5,7 @@ weight: 788
 sidebar_multicard: true
 icon: docs
 hide: true
+draft: true
```
what does `draft: true` do?  It seems odd to have this doc (which would be our `latest` if we could do that) marked as draft.  In any event, will this source directory continue to exist if we are going to put the 1.0.x docs at root level?

StackScribe commented:
For the `description:` for "Previous releases", should we just say "Documentation for previous releases of Keptn"?

StackScribe commented:
We should set hide:true for the content/docs/roadmap/index.md file to remove it from the published docs set.

StackScribe commented:
"Integrations" is under "Previous releases" when it should be at root level and before "Previous releases"

StackScribe commented:
The sidebar expansion when I'm in "Previous versions" doesn't look right -- the other sections are nested under "Explore Keptn" and only a couple are listed -- maybe that is because there is still so much stuff in the left frame?  And when I click on an individual release under "Previous releases", it's taking me back to the bogus 0.19.x docs that are still at the root level -- that may be a vestige of you not being done with this.  I guess I need to wait until you complete the implementation.

aepfli commented:
- changed `integrations`
- changed `description`
- changed `roadmap`

I added another version with previous releases, the only downside is, now previous releases is always showing - no easy way currently of hiding it. and my mind is not in the state of finding a solution

StackScribe commented:
Oh ugh!  This is not good.  I wanted "Previous Releases" to be a walled-off little area.  Maybe we need to just abandon this whole scheme until we can replace the doc build tools -- we'll just show 1.0 LTS above the 0.19.x set and let the 0.x.x docs point to the common Concepts, Installation, etc.  I haven't made that many changes and we may just need to live with it.

The one other possibility is to essentially "freeze" the  0.x.x landing page and publish it on a different URL to which we link from the 1.0.x landing page.  Then we just remove (hide) all the 0.x.x docs from the main landing page.  How hard would it be to implement that?  We could even copy the repo as it is now into a new repo.  I don't expect to make many changes to the 0.x.x docs going forward.  If we go that way, we can go ahead and do that any time -- I am certainly not going to do any work on the 0.x.x docs between now and Monday.  So the new repo with the new landing page (keptn.sh/docs/previous-releases maybe?) could just be sitting there.  Then we take my PR to publish the 1.0 docs.  It sets hide:false for the 1.0.x tree and I can set hide:true for all the 0.x.x trees and anything else we want to lose (like the Roadmap) and voila! 

You've put in a whole lot of effort on this and it may be time to just cut our losses here!  I actually kind of like the idea, frankly -- unless it's a lot harder to implement than I think it is...

aepfli commented:
closing for now, we can re-evaluate later
